pkg/build/builder/common.go

```
@@ -81,3 +84,47 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	glog.Infof("Running post commit hook with image %s ...", imageID)
+
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	cpuPeriod := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	cpuQuota := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	memory := r.read("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if r.err != nil {
```

@rhcarvalho I questioned "readability" argument you gave, not correctness. This hides the error and delegate it to a struct function. While that is Go-compliant and "nicer", I don't think it improves readability, IMHO it makes the code harder to read as you need to go understand the struct function first to know how errors are handled...